File: JUNE 24/12 June/75. Sort Colors LC.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        int countZero=0, countOne=0, countTwo=0;
        for(int num:nums)
        {
            if(num==0)
                countZero++;
            else if(num==1)
                countOne++;
            else
                countTwo++;
        }
        int k=0;
        while(countZero--)
            nums[k++] = 0;
        while(countOne--)
            nums[k++] = 1;
        while(countTwo--)
            nums[k++] = 2;
    }
};

File: MAY 2025/17 May/75. Sort Colors LC.java
class Solution {
    public void sortColors(int[] nums) {
        int red=0, white=0, blue=0;
        for(int num:nums)
        {
            if(num==0)
                red++;
            else if(num==1)
                white++;
            else
                blue++;
        }
        int N = nums.length, index=0;
        while(red>0)
        {
            nums[index++] = 0;
            red--;
        }
        while(white>0)
        {
            nums[index++] = 1;
            white--;
        }
        while(blue>0)
        {
            nums[index++] = 2;
            blue--;
        }
    }
}
